### PR TITLE
Pose estimation models that are suitable for mobile inference

### PR DESCRIPTION
**Pull request by hetong007**
This is a PR for a set of new pose estimation models.

Basically I replace the `deconv` layers in `simple_pose` models with `pixel_shuffle`.
Under the same training process, it achieves great speed up with almost no loss of accuracy.

Additionally, I tried to incorporate the `DSNT` operator to speed up the heatmap to coordinate operation, based on this implementation (https://github.com/YuliangXiu/MobilePose-pytorch) and the reference paper (https://arxiv.org/abs/1801.07372). However the final training accuracy is not comparable (66 vs 52). Now I only use the dsnt module in inference and the accuracy is not as good as finding the max coords deterministically yet. Potential fix will come in a later PR.

Pre-trained models will be added soon. CI is expected to fail at this moment.

Expected follow up works:
- Inference with DSNT module that gets similar accuracy
- Deployment tutorials with TVM team

### CODE REVIEW

File: gluoncv/model_zoo/simple_pose/mobile_pose.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+from __future__ import division
+
+from mxnet import initializer
+from mxnet.gluon import nn
+# from mxnet.gluon.nn import BatchNorm
+from mxnet.gluon.block import HybridBlock
+from mxnet.context import cpu
+
+from ...nn.block import DUC, BatchNormCudnnOff
```
[the remark this answers is not in the document]
Comment (hetong007) [reply gluoncv/model_zoo/simple_pose/mobile_pose.py]: I haven't tried it. Will try with one model and see.

File: gluoncv/utils/metrics/coco_keypoints.py [hunk 2]
```diff
@@ -128,6 +128,9 @@ def update(self, preds, maxvals, score, imgid, *args, **kwargs):
             if count > 0:
                 kpt_score /= count
             rescore = kpt_score * score[idx].asscalar()
+            fixed_score = kwargs.get('fixed_score', False)
```
Comment (bryanyzhu) [inline gluoncv/utils/metrics/coco_keypoints.py]: Can I know what is `fixed_score` and the reason for fixing rescore to 2? Thank you.

Comment (hetong007) [reply gluoncv/utils/metrics/coco_keypoints.py]: Good catch, it was for DSNT modules which only outputs the coordinates but no score. Now I fix the score outside the function call so it should be removed. Thanks.

**Comment by mli** [comment]
Job PR-979-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-979/5/index.html 
Code coverage of this PR:  vs. Master:

**Comment by bryanyzhu** [comment]
Great, looks good to me.

**Comment by hetong007** [comment]
@zhreshold switching back to normal BN yields very similar results. 

Another thing, is there anything I need to do to update the speed/acc plot on the model zoo page?
